build.sbt

```diff
@@ -200,7 +217,18 @@ lazy val io = project
     OsgiKeys.additionalHeaders := Map("-removeheaders" -> "Include-Resource,Private-Package"),
     osgiSettings
   )
-  .dependsOn(coreJVM % "compile->compile;test->test")
+  .jvmSettings(
+    Test / fork := true,
+    libraryDependencies += "com.github.jnr" % "jnr-unixsocket" % "0.38.8" % Optional
+  )
+  .jsSettings(
+    scalaJSLinkerConfig ~= (_.withModuleKind(ModuleKind.CommonJSModule)),
+    Compile / npmDependencies += "@types/node" -> "16.0.0",
+    useYarn := true,
+    stOutputPackage := "fs2.js",
```

Comment:
Whatever you want, really! Happy to make it `jsdeps`.

Reply:
Oh, the reason I didn't put it in `io` ... the JS bindings could be useful outside of `io`?

Reply:
After pondering a bit: I think the package should be `fs2.internal.jsdeps` and we should not expose these bindings in public APIs. There are _lots_ of them, and because they are auto-generated by [Scalably Typed](https://scalablytyped.orgl) which is technically still beta and they interact with the Typescript ecosystem, I'm just scared whether we will be able to control their binary evolution. The primary purpose for adding these bindings was to help me build the internal implementations safely, not for public consumption.

For the couple places we do expose public APIs of JS-native objects, we either:
1. provide our own bindings (that we control bincompat of), or
2. just accept `js.Any` (or a newtype wrapper around that) in the spirit of BYOB (bring-your-own-bindings)

@mpilquist thoughts?

Reply:
If you want to use ST for this the way forward would be to use https://scalablytyped.org/docs/library-developer to generate a library which could be published independently of fs2 releases. If you use very little I'd just copy/paste the necessary parts to be completely in control.

Reply:
@oyvindberg thanks for chiming in! I did find that guide very helpful when setting up the current build.

I'm still inclined to use the ST facades as internal API (it would be awesome if there's a way to automatically mark them as package-private?). I do appreciate the (continued) automation ST provides and would prefer not to chase down the copy/paste necessary to take control. Furthermore, the public APIs we provide in their place can be much more idiomatic: `Chunk[Byte]` instead of `Buffer`, callbacks `A => F[B]` instead of `(A, B => Unit) => Unit`, etc.

Reply:
Aha, I just jumped in and assumed you used the default application mode - apologies.
In that case if it's fast enough to not upset anyone and marked as internal it's probably a good idea.

By package private, do you mean `private[fs2]` for every generated thing? I'd say that would be easy enough to add

Reply:
@oyvindberg No reason to apologize for docs so good that I don't need your help :wink:

Glad you agree! Yes, that's exactly what I mean by package private :)